Add option for custom HipChat installs

ciwchris opened the pull request:
A optional HipChat BaseUrl has been added to allow publishing notifications to a user specified URL, instead of only to a hosted HipChat install.

ciwchris committed:
Add option for custom HipChat installs

A optional HipChat BaseUrl has been added to allow publishing notifications to a user specified URL, instead of only to a hosted HipChat install.

ciwchris commented:
I don't know how to install a locally generated Seq App, so I wasn't able to completely test my change.

kristofferlindvall commented on Seq.App.HipChat/HipChatReactor.cs:
```diff
@@ -52,7 +60,7 @@ public class HipChatReactor : Reactor, ISubscribeTo<LogEventData>
             using (var client = new HttpClient())
             {
-                client.BaseAddress = new Uri("https://api.hipchat.com/v2/");
+                client.BaseAddress = new Uri(HipChatBaseUrl ?? DefaultHipChatBaseUrl);
```
Thanks for your contrib!

I'm unsure about Seq's behaviour, but it would feel more safe using string.IsNullOrWhitespace to check for empty value.

Check the following comment for how to run locally: https://github.com/stayhard/Seq.App.HipChat/pull/1#issuecomment-116478104

ciwchris committed:
Update HipChat URL check to IsNullOrWhiteSpace

It is unknown exactly how Seq sets optional properties. Therefore, to be safe, the optional HipChat base URL is now checking for both null and whitespace, instead of using the coalescing operator.

ciwchris commented:
Updated. Thanks for the tip on using a local nuget feed, I don't know how I missed it!
